db/migrate/20200211225056_modify_offers_table.rb

```diff
@@ -0,0 +1,87 @@
+class ModifyOffersTable < ActiveRecord::Migration[5.2]
+  def up
+    change_table :offers, bulk: true do |t|
+      t.rename(:photography_cents, :photography_info)
+      t.rename(:shipping_cents, :shipping_info)
+      t.rename(:insurance_cents, :insurance_info)
+      t.rename(:other_fees_cents, :other_fees_info)
+
+      t.change(
+        :photography_info,
+        :string,
+        using:
+          'CASE WHEN (photography_info IS NOT NULL) ' \
+            "THEN format('%s %s', TRIM(to_char(photography_info / 100, '99999')), currency) END"
+      )
+      t.change(
+        :shipping_info,
+        :string,
+        using:
+          'CASE WHEN (shipping_info IS NOT NULL) ' \
+            "THEN format('%s %s', TRIM(to_char(shipping_info / 100, '99999')), currency) END"
+      )
+      t.change(
+        :insurance_info,
+        :string,
+        using:
+          'CASE WHEN (insurance_info IS NOT NULL) ' \
+            "THEN format('%s %s', TRIM(to_char(insurance_info / 100, '99999')), currency) " \
+            'WHEN (insurance_percent IS NOT NULL) ' \
+            "THEN TRIM(TO_CHAR(insurance_percent * 100, '99D00%')) END"
+      )
+      t.change(
+        :other_fees_info,
+        :string,
+        using:
+          'CASE WHEN (other_fees_info IS NOT NULL) ' \
+            "THEN format('%s %s', TRIM(to_char(other_fees_info / 100, '99999')), currency) " \
+            'WHEN (other_fees_percent IS NOT NULL) ' \
+            "THEN TRIM(TO_CHAR(other_fees_percent * 100, '99D00%')) END"
+      )
+
+      t.column(:deadline_to_consign, :string)
+
+      t.remove(:insurance_percent)
+      t.remove(:other_fees_percent)
+    end
+  end
+
+  def down
+    change_table :offers, bulk: true do |t|
+      t.column(:insurance_percent, :float)
+      t.column(:other_fees_percent, :float)
+
+      t.remove(:deadline_to_consign)
+
+      t.change(
+        :photography_info,
+        :integer,
+        using:
+          "CASE WHEN (photography_info ~ '^\d+$') THEN photography_info::integer ELSE null END"
```

Comment:
While this _could_ preserve data when reverting the migration, it wouldn't actually match any of the data that was created in the `up`. These are mostly here because if we don't do _something_, it'll try to convert all the just-plain-text values to integer and fail.

Reply:
FWIW, I feel that you've gone above and beyond here - I would have done something like this:

```ruby
raise ActiveRecord::IrreversibleMigration, "Can't go back!"
```

Your way is certainly better. I guess I'm just noting that it's a valid strategy to consider throwing up one's hands on the `down` method. 😝 

But again bravo for being extremely thorough!! 🤗 💯

Reply:
I thought I _was_ throwing up my hands on the `down` method 😅